uportal-war/src/main/java/org/apereo/portal/layout/dlm/DistributedLayoutManager.java

```diff
@@ -452,9 +453,80 @@ public IUserLayoutNodeDescription addNode(
 
             return node;
         }
+
+        if (LOG.isTraceEnabled()) {
+            LOG.trace("Attempted to add node in impermissible location "
+                    + "with parent " + parentId + " and next sibling " + nextSiblingId
+                    + ". Made no layout change.");
+        }
+
         return null;
     }
 
+    /*
+     * DLM implementation adds to first allowable position among existing children of the parent
+     * node, if any.
+     */
+    public IUserLayoutNodeDescription addNodeInAnyOrder(
+        final IUserLayoutNodeDescription nodeToInsert, final String parentId) {
+
+        final IUserLayoutNodeDescription parentNode = getNode(parentId);
+
+        final String firstAllowableNextSiblingId =
+            firstAllowableNextSiblingId(parentNode, nodeToInsert);
+
+        return addNode(nodeToInsert, parentId, firstAllowableNextSiblingId);
+    }
+
+    /**
+     * Compute the siblingId representing the earliest allowable placement for adding a child node.
+     * @param parent potential parent of the node to be added
+     * @return siblingId representing earliest allowable nextSiblingId for adding a node,
+     * or blank if no such nextSiblingId
+     * @since 5.0.0
+     */
+    private String firstAllowableNextSiblingId(
```

Comment:
Suggest `calculateFirstAllowableNextSiblingId` or something akin.  Adding a verb to the name will help with clarity.